PatternPal/PatternPal.Core/CheckResult.cs

```diff
@@ -70,4 +95,15 @@ internal class NodeCheckResult : ICheckResult
     /// Behavior of the collection of sub-<see cref="ICheck"/>s.
     /// </summary>
     public CheckCollectionKind CollectionKind { get; init; } = CheckCollectionKind.All;
+
+    /// <inheritdoc />
+    public required int DependencyCount { get; init; }
+
+    /// <inheritdoc />
+    public required INode ? MatchedNode { get; init; }
+
+    /// <summary>
+    /// All sub-<see cref="ICheckResult"/>s are the result of a <see cref="NodeCheck{TNode}"/>.
+    /// </summary>
+    public bool NodeCheckCollectionWrapper { get; set; }
```

Comment:
It is not quite clear for me what this is or how it is evaluated, could you elaborate on that?